mpi-proxy-split/cartesian.h

```diff
@@ -1,6 +1,8 @@
 #ifndef CARTESIAN_H
 #define CARTESIAN_H
 
+#ifdef SINGLE_CART_REORDER
```

Comment:
I don't think you need to comment everything out within this header file. You can take care of that while including this header file. It means to include this header file only when the macro is defined.